### PR TITLE
fix: fail to output json with batch option 'aws'

### PR DESCRIPTION
**Pull request by shisama**
duck fails to output json report when error occurs.
The reason is that it cannot parse message including 'logUrl' to JSON.

Error message is like the below. In this case, the line `: https://ap-northeast-1.console...` is logUrl.
```
Unexpected non-JSON error: /opt/nodejs/node_modules/google-closure-compiler-linux/compiler --rewrite_polyfills=true ...
[{"level":"error","description":"The result of a goog.define call must be assigned as an isolated statement.", ...},{"level":"info","description":"70 error(s), 0 warning(s)"}]

: https://ap-northeast-1.console.aws.amazon.com/cloudwatch/home?region=ap-northeast-1#logEventViewer:group=%2Faws%2Flambda%2Ffaast-b0d763dd-d988-4d38-a357-485f28d8ef5e;stream=2020%2F01%2F30%2F%5B%24LATEST%5D12d386dbf1fe4ca9933e07671ca6956c;filter=%22575f34df-23ac-47a3-b0d9-9bb5f8a081e0%22

```

`logUrl` has been included since v3.0.14.
see the commit: https://github.com/faastjs/faast.js/commit/4c65cffa1e522e886f3948ee85b1943925e78281
So, this problem does not happen before v3.0.13.

[FaastError document](https://faastjs.org/docs/api/faastjs.faasterror) indicates the below.
`The logUrl will be appended to the end of FaastError.message and FaastError.fullStack as well.`

This PR deletes 'logUrl' from error message to be able to parse it to JSON.

### CODE REVIEW

**Comment by teppeis** [comment]
Note: the colon separated error message format for nested error objects is created by [verror](https://github.com/joyent/node-verror).

**Comment by shisama** [comment]
@teppeis 
I have some ideas. It was hard to choose one for me because I didn't know what is better. If you don't mind, please see the ideas and give me your idea if you have.

**Comment by koba04** [comment]
@shisama Thank you for your great work and proposals!!! That's awesome 👍 

I prefer the approach of 4. it seems the most rigid approach for me because the message might be added other than `logUrl` in the future release.
https://gist.github.com/shisama/d182500cf8f70344fea36a4ecc355e10#gistcomment-3164710

I feel that we should isolate code for faastjs, but it would require a lot of works.
So it would be better to work on it as another task.

But getting a `command` string and `CompilerErrorItem[]` has already messed up, so I feel that it would be nice if the implementation is separated as a function like the following.

```ts
const { command, items } = parseErrorReason(result.reason, config);
:
function parseErrorReason(
  // FIXME
  reason: unknown,
  config: DuckConfig
): { command: string; items: CompileErrorItem[] } {
  const { message: stderr, info } = reason as CompilerError & {
    info?: Record<string, any>;
  };
  const message = config.batch === "aws" && info ? info.message : stderr;
  const [command, , ...messages] = message.split("\n");
  return { command, items: JSON.parse(messages.join("\n")) };
}
```

**Comment by koba04** [comment]
@shisama Thank you! Could you fix the conflicts?

File: src/commands/buildJs.ts [hunk 1]
```diff
@@ -150,6 +150,16 @@ async function waitAllAndThrowIfAnyCompilationsFailed(
     throw new BuildJsCompilationError(reasons, results.length);
   }
   return reasons;
+
+  function parseErrorReason(
+    reason: CompilerError & { info?: Record<string, any> },
+    config: DuckConfig
+  ): { command: string; items: CompileErrorItem[] } {
+    const { message: stderr, info } = reason;
+    const message = config.batch === "aws" && info ? info.message : stderr;
```
Comment (teppeis) [inline src/commands/buildJs.ts]: This issue occurs not only with `--batch aws` but also with `--batch local`.

Comment (shisama) [reply src/commands/buildJs.ts]: @teppeis Thanks. I fixed this. https://github.com/cybozu/duck/pull/378/commits/b8e1583d81489ebaa3f0c51a13550f9f143ac4ab

**Comment by shisama** [comment]
@koba04 Thanks. I rebased this and resolved the conflicts.

File: src/commands/buildJs.ts [hunk 2]
```diff
@@ -142,25 +144,41 @@ async function waitAllAndThrowIfAnyCompilationsFailed(
...
     throw new BuildJsCompilationError(reasons, results.length);
   }
   return reasons;
+
+  function parseErrorReason(
+    reason: CompilerError & { info?: Record<string, any> },
+    config: DuckConfig
+  ): { command: string; items: CompileErrorItem[] } {
+    const { message: stderr, info } = reason;
+    const message = isBatchMode(config.batch) && info ? info.message : stderr;
```
Comment (koba04) [inline src/commands/buildJs.ts]: @shisama 
According to https://github.com/cybozu/duck/pull/378#discussion_r378350303, this issue would occur when `duck` use faastjs to build JS.

duck decides whether to use faastjs or not by the following condition.
https://github.com/cybozu/duck/blob/b8e1583d81489ebaa3f0c51a13550f9f143ac4ab/src/commands/buildJs.ts#L37
So I guess that you can use the same condition for this.

Comment (shisama) [reply src/commands/buildJs.ts]: @koba04 
Thank you. I fixed it. If you think that we don't need to check if 'info' exists, please tell me again.

**Comment by koba04** [comment]
@teppeis Could we merge this PR? Your review status isn't in approval yet.

**Comment by teppeis** [comment]
@shisama @koba04 LGTM 👍

**Pull request merged**